src/modules/pmc/src/lib/PmcBase.cpp

```diff
@@ -561,3 +599,114 @@ std::string PmcBase::Trim(const std::string& str, const std::string& trim)
 {
     return TrimStart(TrimEnd(str, trim), trim);
 }
+
+std::vector<std::string> PmcBase::ListFiles(const char* directory, const char* fileNameExtension)
+{
+    struct dirent* entry = nullptr;
+    DIR* directoryStream = nullptr;
+    int extensionLength = (nullptr != fileNameExtension) ? strlen(fileNameExtension) : 0;
+    char* fileName;
+    int fileNameLength;
+    char* lastCharacters;
+    std::vector<std::string> result;
+
+    directoryStream = opendir(directory);
+    if (nullptr != directoryStream)
+    {
+        while ((entry = readdir(directoryStream)))
+        {
+            fileName = entry->d_name;
+            if ((0 == strcmp(fileName, ".")) || (0 == strcmp(fileName, "..")))
+            {
+                continue;
+            }
+
+            if (nullptr == fileNameExtension)
+            {
+                result.push_back(fileName);
+            }
+            else
+            {
+                fileNameLength = strlen(fileName);
+                if (fileNameLength >= extensionLength)
+                {
+                    lastCharacters = &fileName[fileNameLength-extensionLength];
```

Comment:
>fileName[fileNameLength-extensionLength] [](http://example.com/codeflow?start=38&length=40)

Check for size

Reply:
it is checked in the line 631 with this: `if (fileNameLength >= extensionLength)`